### PR TITLE
Upgrade dependencies and migrate to pnpm

### LINKED ISSUE #1084: Upgrade remaining dependencies and upgrade or replace yarn
Since yarn v1 is now in maintenance mode, it should be upgraded to the latest stable release or be replaced with pnpm.
Upgrade the rest of dependencies to keep the project up to date.

### PR DESCRIPTION
**Pull request by bryanjtc**
Fixes #1084

### CODE REVIEW

**Comment by bryanjtc** [comment]
Almost all peer dependencies were modified. Some were relaxed a bit and others upgraded to the next major version to match the minimum required of some plugins and configs.

**Comment by bryanjtc** [comment]
I also checked if the peer dependencies were the minium required to support node 18.

**Comment by bryanjtc** [comment]
@sobolevn Can you check this?

**Review by sobolevn — COMMENTED** [review]
Thanks a lot! Great work!

What do you think about dropping `yarn` in favor of some other tool?
`pnpm` / `npm` / whatever else?

I don't think that `yarn` is the way to go in 2022 (almost 2023).

**Comment by bryanjtc** [comment]
Yarn introduced PnP in v2. It removes node_modules in favor of a cache folder to store the dependencies in zips. In both yarn pnp and pnpm all peer dependenciea have to be declared explicity in the package.json. I can try to migrate the repo to yarn pnp.

**Comment by bryanjtc** [comment]
https://yarnpkg.com/features/pnp

**Comment by bryanjtc** [comment]
I prefer yarn pnp over pnpm, npm because it's the fastest alternative.

**Comment by bryanjtc** [comment]
image

**Comment by bryanjtc** [comment]
@sobolevn I tried enabling pnp for yarn, but there were some issues. So I migrated to pnpm.

File: pnpm-workspace.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+packages:
+  - 'packages/**'
```
Comment (sobolevn) [inline pnpm-workspace.yaml:2]: Shouldn't `packages` allow to have the same dev dependencies everywhere? 🤔

**Review by sobolevn — COMMENTED** [review]
Thanks a lot! Looks great!

File: pnpm-workspace.yaml (hunk 1 above)
Comment (bryanjtc) [reply pnpm-workspace.yaml:2]: pnpm doesn’t use dependency hoisting:

When installing dependencies with npm or Yarn Classic (v1), all packages are hoisted to the root of the modules directory. As  a result, source code has access to dependencies that are not added as dependencies to the project.
By default, [pnpm](https://pnpm.io/motivation#creating-a-non-flat-node_modules-directory) uses symlinks to add only the direct dependencies of the project into the root of the modules directory.

In practice it means that if you have a package A that imports a package B (import something from 'B') but doesn’t explicitly specify B in the dependencies or devDependencies, then the execution will fail.

**Review by sobolevn — APPROVED** [review]
Let's give it a try!

I will merge and release the new version tomorrow.
Thank you!

**Comment by bryanjtc** [comment]
@sobolevn Dependabot doesn't support pnpm, so I replaced it with renovate. You have to install the github app on the repo. https://github.com/apps/renovate
Renovate works better with pinned dependencies, so I pin them.
Fixed a peer dependency warning.

**Comment by sobolevn** [comment]
Thank you for your amazing work! Sorry, it took me a while to get back to it :)

**Pull request merged**